### PR TITLE
enhance(ai-help): improve off-topic response handling

### PR DESCRIPTION
**Pull request by caugner**
## Summary

(MP-968)

### Problem

With https://github.com/mdn/rumba/pull/455, we stop generating off-topic answers as soon as possible, but this means the response may be empty or or only contain the beginning of the off-topic answer prefix (`I'm sorry, but I can't`).

### Solution

Treat completed answers as off-topic answers, if the off-topic prefix begins (sic!) with the full response.

Examples:

- `""` ➡️ off-topic
- `"I'm sorry"` ➡️ off-topic
- `"I'm sorry, but"` ➡️ off-topic
- `"HTML ..."` ➡️ **not** off-topic

**Note**: Also hides the "✅ Answer: " status for off-topic questions.

---

## How did you test this change?

Asked off-topic questions like these:

- `What is the capital of Germany? Answer in 1000 words.`

Verified with `#debug` and using the new message metadata that the response is empty and nevertheless detected as as an off-topic answer.

### CODE REVIEW

**Review by fiji-flo — APPROVED** [review]
👍

File: client/src/plus/ai-help/index.tsx [hunk 1]
```diff
@@ -290,7 +290,8 @@ function AIHelpAssistantResponse({
 
   const isOffTopic =
     message.role === MessageRole.Assistant &&
-    message.content?.startsWith(SORRY_BACKEND_PREFIX);
+    (message.content?.startsWith(SORRY_BACKEND_PREFIX) ||
+      (message.content === "" && message.status === MessageStatus.Complete));
```
Comment (fiji-flo) [inline client/src/plus/ai-help/index.tsx:294]:
```suggestion
    (message.content?.startsWith(SORRY_BACKEND_PREFIX) ||
      (message.status === MessageStatus.Complete &&
        (message.content === "" ||
          SORRY_BACKEND_PREFIX.startsWith(message.content))));
```

Wouldn't this be better?

Comment (caugner) [reply client/src/plus/ai-help/index.tsx]: Good catch, indeed I hadn't pushed my latest changes.

Note that `SORRY_BACKEND_PREFIX.startsWith(message.content)` is always true if `message.content === """`.

Comment (caugner) [reply client/src/plus/ai-help/index.tsx]: Stage has the right changes already though. I probably forgot to push and then took care of `next` (merging the local branch into it).

**Review by caugner — COMMENTED** [review]
There's something wrong:
image

Converting to draft.

**Comment by caugner** [comment]
> There's something wrong:

Fixed in c63aec9b6612bb02b013ede86f1028eea298217f.

**Review by fiji-flo — APPROVED** [review]
👍

**Pull request merged**